worker/src/core/loaders/message-worker/index.ts

```diff
@@ -248,6 +256,111 @@ const enqueueAndSend = async (): Promise<void> => {
   }
 }
 
+// to refactor to use orm
+const sendFinalizedNotification = (campaignId: number): void => {
+  void connection
+    .query(
+      "SELECT json_build_object('id', c.id," +
+        "'campaign_name', c.name," +
+        "'created_at', c.created_at," +
+        "'unsent_count', s.unsent," +
+        "'error_count', s.errored," +
+        "'sent_count', s.sent," +
+        "'invalid_count', s.invalid," +
+        "'notification_email', u.email," +
+        "'halted', c.halted) as result FROM campaigns c INNER JOIN statistics s ON c.id=s.campaign_id INNER JOIN users u ON c.user_id=u.id WHERE c.id=:campaign_id_input;",
+      {
+        replacements: { campaign_id_input: campaignId },
+        type: QueryTypes.SELECT,
+      }
+    )
+    .then(async (result) => {
+      const jsonRes = get(result, '[0].result') || {}
+      const {
+        id: campaignId,
+        campaign_name: campaignName,
+        created_at: createdAt,
+        unsent_count: unsentCount,
+        error_count: errorCount,
+        sent_count: sentCount,
+        invalid_count: invalidCount,
+        notification_email: notificationEmail,
+        halted: halted,
+      } = jsonRes
+
+      let mail: MailToSend | void
```

Comment:
this should be `null` instead of `void` (see https://www.tutorialsteacher.com/typescript/typescript-void)

actually, when will `mail` not be falsy? there should always be a mail sent? so the nested`if (mail)` statement can be removed?

Reply:
this is keeping consistent with how it has been implemented elsewhere...

Reply:
sorry, to elabroate, a function can return `void` (i.e. it doesn't return anything), but `void` is not a valid type in TypeScript

> this is keeping consistent with how it has been implemented elsewhere...
ohh which parts? haha should flag so that we can refactor those away